src/base/ReactorStructs.sol

```diff
@@ -34,12 +32,24 @@ struct OrderInfo {
     uint256 deadline;
 }
 
-/// @notice Every RelayOrder input is defined by a token, recipient,
-/// and amounts that define the start and end amounts on the decay curve.
-/// @dev These values are signed by the user. address(0) will set the recipient at run-time to the passed in feeRecipient value.
+/// @notice Every RelayOrder input is defined by a token, amount, and recipient,
+/// @dev These values are signed by the user
 struct Input {
     address token;
+    uint256 amount;
     address recipient;
+}
+
+/// @notice A RelayOrder can specify an increasing fee over time to be paid
+struct FeeEscalator {
+    address token;
     uint256 startAmount;
-    uint256 maxAmount;
+    uint256 endAmount;
+    // The time at which the fee starts to increase
+    uint256 startTime;
+    // The time at which the fee becomes static
+    uint256 endTime;
+    // The address to which the fee will be paid
+    // Note that address(0) will send the fee to the recipient specified by the filler of the order
+    address recipient;
```

Comment:
whats the need for specifying this here? afaiu this is the user-specified struct, when would the user ever specify non-zero value here?

Reply:
if they want to restrict it to a filler (ie an exclusive filler, then they would sign a non zero address). 0 address signifies user signing off on anyone receiving the fees

Reply:
context https://github.com/Uniswap/relayer/pull/61#discussion_r1488032643

Reply:
What's the case where that is intended to be used? Other folks can still fill it so it's not really exclusive, they just wouldn't get the fee? If we want to target specific filler it seems that should be a separate strictly defined thing

Reply:
I think its a nice option. Its true can fill it but there is no incentive and the person meant to "fill" still gets the reward.  use case could be like labs run relayer.

I also thought it was just generally nice to have a user explicitly opt in by signing a recipient value that means anyone can take this input.

But if you dont like it we can remove i guess